extensions/native/circuit/src/extension.rs

```diff
@@ -308,6 +314,29 @@ pub(crate) mod phantom {
             Ok(())
         }
     }
+
+    impl<F: PrimeField32> PhantomSubExecutor<F> for NativeHintLoadSubEx {
+        fn phantom_execute(
+            &mut self,
+            _: &MemoryController<F>,
+            streams: &mut Streams<F>,
+            _: PhantomDiscriminant,
+            _: F,
+            _: F,
+            _: u16,
+        ) -> eyre::Result<()> {
+            let payload = match streams.input_stream.pop_front() {
+                Some(hint) => hint,
+                None => {
+                    bail!("EndOfInputStream");
+                }
+            };
+            let id = streams.hint_space.len();
+            streams.hint_space.push(payload);
+            streams.hint_stream.push_back(F::from_canonical_usize(id));
```

Comment:
should we just reset hint stream to `vec![id]` or assert its empty?

Reply:
nvm I see above you explicitly wanted to be able to have multiple id added to hint stream

Reply:
ok this works well with `HINTSTORE_W` since that uses `hint_stream.pop_front`. However, right now we need to inspect the program to ensure the hint stream is clear beforehand. I'm just going to add a `hint_stream.clear()` defensively.